docs/source/ProgrammingGuide/examples/graph_capture.cpp

```diff
@@ -0,0 +1,123 @@
+#include "cublas_v2.h"
+
+#include "Kokkos_Core.hpp"
+#include "Kokkos_Graph.hpp"
+
+#define CHECK_CUBLAS_CALL(call)                           \
+  {                                                       \
+    const auto error_code = call;                         \
+    if(error_code != CUBLAS_STATUS_SUCCESS)               \
+    {                                                     \
+      printf("%s:%d: failure of statement %s: %s (%d)\n", \
+        __FILE__, __LINE__,                               \
+        #call,                                            \
+        cublasGetStatusName(error_code), error_code);     \
+      std::abort();                                       \
+    }                                                     \
+  }
+
+template <typename MatrixType, typename VectorType>
+struct InitializeData
+{
+  MatrixType matrix;
+  VectorType vector;
+
+  template <typename T>
+  KOKKOS_FUNCTION
+  void operator()(const T irow) const
+  {
+      matrix(2 * irow)     = 2 * irow + 1;
+      matrix(2 * irow + 1) = 2 * irow + 2;
```

Comment:
I assume you are using a functor because the parallel_for is in main() and NVCC doesn't let you use a lambda but that is really not great here to have to know what nrows or ncols are.  You have other options (move the computation into some other named function, etc.)
Also, I am not sure why we are using a rank-1 view for the matrix.

Reply:
The matrix is now a rank-2 with layout right enforced.